src/main/scala/is/hail/variant/Genotype.scala

```diff
@@ -929,12 +928,8 @@ class GenericGenotype(val _unboxedGT: Int,
 
   if (_isLinearScale) {
     require(_unboxedGQ == -1)
-    if (_unboxedPX == null)
-      require(_unboxedGT == -1)
-    else {
+    if (_unboxedPX != null)
       require(_unboxedPX.sum == 32768)
-      require(_unboxedGT == Genotype.gtFromLinear(_unboxedPX).getOrElse(-1))
-    }
```

Comment:
this fixes a bug I saw, right? This is a totally separate issue from anything else here.